src/Compilers/CSharp/Portable/Binder/SwitchBinder.cs

```diff
@@ -181,66 +200,35 @@ internal override bool IsLabelsScopeBinder
             }
         }
 
-        private void BuildSwitchLabels(SyntaxList<SwitchLabelSyntax> labelsSyntax, Binder sectionBinder, ref ArrayBuilder<LabelSymbol> labels)
+        private void BuildSwitchLabels(TypeSymbol switchGoverningType, SyntaxList<SwitchLabelSyntax> labelsSyntax, Binder sectionBinder, ArrayBuilder<LabelSymbol> labels, DiagnosticBag tempDiagnosticBag)
```

Comment:
What is the advantage of passing the type here vs. having the code access it as a instance property?  It suggested to me that there were cases where `switchGoverningType` and `this.SwitchGoverningType` were not the same reference.